externalpolicy/text_language/text_language.go

```diff
@@ -69,3 +120,211 @@ func (p *TextConstraintLanguagePlugin) GetNextOperator(expression string) (strin
+		return true
+	}
+	return false
+}
+
+const andsymbol = "&&"
+const orsymbol = "||"
+const and = "AND"
+const or = "OR"
+
+func isAllowedLogicalOpType(s string) bool {
+	if strings.Compare(s, andsymbol) == 0 || strings.Compare(s, orsymbol) == 0 || strings.Compare(s, and) == 0 || strings.Compare(s, or) == 0 {
+		return true
+	}
+	return false
+}
+
+func preprocessConstraintExpression(str string) string {
+	re := regexp.MustCompile(`(?m)"(.*?)"(?m)`)
+
+	// remove space inside string list separate by ", "
+	space := regexp.MustCompile(`,\s+`)
+	s := space.ReplaceAllString(str, ",")
+
+	for _, match := range re.FindAllString(s, -1) {
+		// if find "a b", replace space inside quote with invisiable charactor \a
+		newStr := strings.Replace(match, " ", "\a", -1)
+		s = strings.Replace(s, match, newStr, -1)
+	}
+
+	return s
+}
+
+// 1. == is supported for all types except list of string, which would use 'in'.
+// 2. for numeric types, the operators ==, <, >, <=, >= are supported
+// 3. false and true are the only valid values for a boolean type
+// 4. for string types, a quoted string, inside which is a list of comma separated strings provide acceptable values
+// 5. string values that contain spaces must be quoted
+// 6. for the version type, supported values are a single version or a range of versions in the semantic version format (the same as used for service verions). The == operator implies that the value is a single version. The 'in' operator treats the value as a version range. As with service versions, the version 1.0.0 when treated as a version range is equivalent to the explicit range [1.0.0,INFINITY).
+
+func validateOneConstraintExpression(expression string) (bool, error) {
+	if len(expression) == 0 {
+		return true, nil
+	}
+
+	pieces := strings.Split(expression, " ")
+	if len(pieces) < 3 {
+		return false, errors.New(fmt.Sprintf("found %v token(s), expecting 3 in an expression %v, expected form is <property> == <value> in constraint expression", len(pieces), expression))
+	}
+
+	compOp := pieces[1]
+	value := pieces[2]
+
+	// if will failed on case when string values that contain spaces but not quoted (starting from 2nd interation)
+	if !isAllowedComparisonOpType(pieces[1]) {
+		return false, errors.New(fmt.Sprintf("Expression: %v should contain valid comparison operator - wrong operator %v", expression, pieces[1]))
+	}
+
+	if canParseToFloat(value) || canParseToInteger(value) {
+		if strings.Compare(compOp, doubleequalto) == 0 || strings.Compare(compOp, equalto) == 0 || strings.Compare(compOp, lessthan) == 0 || strings.Compare(compOp, greaterthan) == 0 || strings.Compare(compOp, lessthaneq) == 0 || strings.Compare(compOp, greaterthaneq) == 0 {
+			return true, nil
+		}
+		return false, errors.New(fmt.Sprintf("1. Comparison operator: %v is not supported for numeric value: %v", compOp, value))
```

Comment:
I think I understand why you included the prefix numbers in the error msg, e.g 1., 2. etc but we dont do that anywhere else in the code. It looks like there is enough context in the rest of msg to know the msg is coming from, I think you could remove the numbers.